### ISSUE TITLE
Large regular expression matches cause a segfault

### ISSUE THREAD

**Issue by elyscape** [stedolan/jq#1370]
Here's an interesting bug: when a regular expression with a large repeated number of characters (i.e. anything followed by `+`, `*`, `{count}`, `{min,}`, or `{min,max}`) is attempted against a string, `jq` segfaults:
```shellsession
scapeless:~ elyscape$ VAL="$(echo -n '"' ; for num in {1..200} ; do echo -n 'a' ; done ; echo '"')"
scapeless:~ elyscape$ echo "$VAL"
"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
scapeless:~ elyscape$ # dot matching:
scapeless:~ elyscape$ echo "$VAL" | jq 'match(".*")'
Segmentation fault: 11
scapeless:~ elyscape$ echo "$VAL" | jq 'match(".+")'
Segmentation fault: 11
scapeless:~ elyscape$ echo "$VAL" | jq 'match(".{159}")'
Segmentation fault: 11
scapeless:~ elyscape$ echo "$VAL" | jq 'match(".{158}")'
{
  "offset": 0,
  "length": 158,
  "string": "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
  "captures": []
}
scapeless:~ elyscape$ echo "$VAL" | jq 'match(".{1,80}")'
Segmentation fault: 11
scapeless:~ elyscape$ echo "$VAL" | jq 'match(".{1,79}")'
{
  "offset": 0,
  "length": 79,
  "string": "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
  "captures": []
}
scapeless:~ elyscape$ # exact character matching:
scapeless:~ elyscape$ echo "$VAL" | jq 'match("a*")'
Segmentation fault: 11
scapeless:~ elyscape$ echo "$VAL" | jq 'match("a+")'
Segmentation fault: 11
scapeless:~ elyscape$ echo "$VAL" | jq 'match("a{159}")'
Segmentation fault: 11
scapeless:~ elyscape$ echo "$VAL" | jq 'match("a{158}")'
{
  "offset": 0,
  "length": 158,
  "string": "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
  "captures": []
}
scapeless:~ elyscape$ echo "$VAL" | jq 'match("a{1,80}")'
Segmentation fault: 11
scapeless:~ elyscape$ echo "$VAL" | jq 'match("a{1,79}")'
{
  "offset": 0,
  "length": 79,
  "string": "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
  "captures": []
}
```
If you know something vaguely unique that will happen in the middle of the string you're matching, you can avoid a segfault by breaking up the repeated parts:
```shellsession
scapeless:~ elyscape$ VAL="$(echo -n '"' ; for char in {a..t} ; do echo -n "123456789${char}" ; done ; echo '"')"
scapeless:~ elyscape$ echo "$VAL"
"123456789a123456789b123456789c123456789d123456789e123456789f123456789g123456789h123456789i123456789j123456789k123456789l123456789m123456789n123456789o123456789p123456789q123456789r123456789s123456789t"
scapeless:~ elyscape$ echo "$VAL" | jq 'match(".*")'
Segmentation fault: 11
scapeless:~ elyscape$ echo "$VAL" | jq 'match(".*f.*")'
{
  "offset": 0,
  "length": 200,
  "string": "123456789a123456789b123456789c123456789d123456789e123456789f123456789g123456789h123456789i123456789j123456789k123456789l123456789m123456789n123456789o123456789p123456789q123456789r123456789s123456789t",
  "captures": []
}
```

## tl;dr
The limits beyond which segfaults happen appear to be:

Regex|Max safe length|Segfault length
-|-|-
`.+`/`.*`|159|160
`.{count}`|158|159
`.{min,}`|159 + `min`|160 + `min`
`.{0,max}`|79|80
`.{min,max}`|78 + `min`|79 + `min`

**Comment by pkoppstein** [comment]
@elyscape - I've used several of your examples with various versions of jq on a Mac and cannot reproduce the problem you describe.  Could you please give further details about the relevant version or versions of jq (jq -V) you're using, and some information about the environment, e.g. `uname -a` or as appropriate.

**Comment by elyscape** [comment]
Currently:
```shellsession
scapeless:~ elyscape$ jq -V
jq-1.5
scapeless:~ elyscape$ uname -a
Darwin scapeless.local 16.4.0 Darwin Kernel Version 16.4.0: Thu Dec 22 22:53:21 PST 2016; root:xnu-3789.41.3~3/RELEASE_X86_64 x86_64
```
I tried this with a build of the current master branch, however, and it happened then as well.

**Comment by pkoppstein** [comment]
For reference:

```
$ uname -a
Darwin Mac-mini.local 13.4.0 Darwin Kernel Version 13.4.0: Mon Jan 11 18:17:34 PST 2016; root:xnu-2422.115.15~1/RELEASE_X86_64 x86_64
$ jq1.5 -V
jq-1.5
$ jq1.5  'match(".*")' input.json
{
  "offset": 0,
  "length": 200,
  "string": "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
  "captures": []
}
```

**Comment by elyscape** [comment]
Strange. Could it be the version of Oniguruma? I'm using 6.1.3.
<details>
<summary>macOS Crash report</summary>

```
Process:               jq [50960]
Path:                  /usr/local/Cellar/jq/1.5_2/bin/jq
Identifier:            jq
Version:               0
Code Type:             X86-64 (Native)
Parent Process:        bash [39089]
Responsible:           jq [50960]
User ID:               501

Date/Time:             2017-03-23 15:00:42.073 -0700
OS Version:            Mac OS X 10.12.3 (16D32)
Report Version:        12
Anonymous UUID:        C67428CF-4922-CC0F-80C8-58C7E0EF8487


Time Awake Since Boot: 9900 seconds

System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x00007fff582a4000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Signal:    Segmentation fault: 11
Termination Reason:    Namespace SIGNAL, Code 0xb
Terminating Process:   exc handler [0]

VM Regions Near 0x7fff582a4000:
    Stack                  00007fff57aa4000-00007fff582a4000 [ 8192K] rw-/rwx SM=PRV  thread 0
--> 
    Submap                 00007fff70000000-00007fffe6200000 [  1.8G] r--/rwx SM=SHM  machine-wide VM submap

Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   libsystem_platform.dylib      	0x00007fffdd93dfcc _platform_memmove$VARIANT$Haswell + 300
1   libonig.4.dylib               	0x00000001079b5439 stack_double + 141
2   libonig.4.dylib               	0x00000001079aff3b match_at + 3294
3   libonig.4.dylib               	0x00000001079b450d onig_search + 2265
4   jq                            	0x000000010796a967 f_match + 1287
5   jq                            	0x0000000107965e47 jq_next + 5706
6   jq                            	0x000000010795f38b process + 37
7   jq                            	0x000000010795ee62 main + 4034
8   libdyld.dylib                 	0x00007fffdd72d255 start + 1

Thread 0 crashed with X86 Thread State (64-bit):
  rax: 0x00007f9fd5808600  rbx: 0x0000000000000140  rcx: 0x0000000000000020  rdx: 0x00000000000009e0
  rdi: 0x00007f9fd580ade0  rsi: 0x00007fff582a3fd0  rbp: 0x00007fff582a1760  rsp: 0x00007fff582a1760
   r8: 0x0000000000003200   r9: 0x00007f9fd5808600  r10: 0x00007fff582a17f0  r11: 0xffffffa07d566e10
  r12: 0x00007f9fd5808600  r13: 0x00007fff582a30f0  r14: 0x00000000000000a0  r15: 0x0000000000000000
  rip: 0x00007fffdd93dfcc  rfl: 0x0000000000010202  cr2: 0x00007fff582a4000
  
Logical CPU:     0
Error Code:      0x00000004
Trap Number:     14


Binary Images:
       0x10795c000 -        0x10798efff +jq (0) <4908EFEE-A437-39F1-B760-8279577DB3B4> /usr/local/bin/jq
       0x10799f000 -        0x1079f2ffb +libonig.4.dylib (0) <5709A0C3-928A-377D-9F6F-EE0E735EBC8D> /usr/local/opt/oniguruma/lib/libonig.4.dylib
       0x11364b000 -        0x113688267  dyld (421.2) <947FC440-80F9-32F7-A773-6FC418FE1AB7> /usr/lib/dyld
    0x7fffdc18a000 -     0x7fffdc18bff3  libSystem.B.dylib (1238) <9CB018AF-54E9-300F-82BE-81FE553C9154> /usr/lib/libSystem.B.dylib
    0x7fffdc2a3000 -     0x7fffdc2a3ff3  libauto.dylib (187) <5BBF6A00-CC76-389D-84E7-CA88EDADE683> /usr/lib/libauto.dylib
    0x7fffdc2c4000 -     0x7fffdc31aff7  libc++.1.dylib (307.4) <BEE86868-F831-384C-919E-2B286ACFE87C> /usr/lib/libc++.1.dylib
    0x7fffdc31b000 -     0x7fffdc345fff  libc++abi.dylib (307.2) <1CEF8ABB-7E6D-3C2F-8E0A-E7884478DD23> /usr/lib/libc++abi.dylib
    0x7fffdce35000 -     0x7fffdd205d97  libobjc.A.dylib (706) <F9AFE665-A3A2-3285-9495-19803A565861> /usr/lib/libobjc.A.dylib
    0x7fffdd64f000 -     0x7fffdd653ff7  libcache.dylib (79) <0C8092D3-600F-3ADD-A036-F225B6CDCA43> /usr/lib/system/libcache.dylib
    0x7fffdd654000 -     0x7fffdd65fff7  libcommonCrypto.dylib (60092.30.2) <B16E29B6-EC8D-3A8F-9A89-DD9CF35F7C4B> /usr/lib/system/libcommonCrypto.dylib
    0x7fffdd660000 -     0x7fffdd667fff  libcompiler_rt.dylib (62) <E992E8D9-037C-3454-A366-A25E4D31D6BB> /usr/lib/system/libcompiler_rt.dylib
    0x7fffdd668000 -     0x7fffdd670fff  libcopyfile.dylib (138) <64E285D9-5485-333B-AEE7-8B0C8FB9275F> /usr/lib/system/libcopyfile.dylib
    0x7fffdd671000 -     0x7fffdd6f4fdf  libcorecrypto.dylib (442.30.20) <2074B932-FD79-30A9-8E90-AF25C49F2AF1> /usr/lib/system/libcorecrypto.dylib
    0x7fffdd6f5000 -     0x7fffdd727fff  libdispatch.dylib (703.30.5) <EA0CC14E-D559-3802-B4B2-0E8C7579AAC4> /usr/lib/system/libdispatch.dylib
    0x7fffdd728000 -     0x7fffdd72dff3  libdyld.dylib (421.2) <6F506653-FFF6-3DB8-84F1-109AE3C52F32> /usr/lib/system/libdyld.dylib
    0x7fffdd72e000 -     0x7fffdd72effb  libkeymgr.dylib (28) <1A318923-1200-3B06-B432-5007D82F195D> /usr/lib/system/libkeymgr.dylib
    0x7fffdd73c000 -     0x7fffdd73cfff  liblaunch.dylib (972.30.7) <15FACC21-079A-3BDF-9AFB-4253EFDEB587> /usr/lib/system/liblaunch.dylib
    0x7fffdd73d000 -     0x7fffdd742fff  libmacho.dylib (894) <A2F38EC1-C37C-3B93-B0E4-36B07C177F8C> /usr/lib/system/libmacho.dylib
    0x7fffdd743000 -     0x7fffdd745ff3  libquarantine.dylib (85) <C1D7749F-5F5F-3BB9-BEFC-1F0B9DA941FD> /usr/lib/system/libquarantine.dylib
    0x7fffdd746000 -     0x7fffdd747ffb  libremovefile.dylib (45) <CD42974E-BE0B-39FC-9BFC-8A7540A04DC6> /usr/lib/system/libremovefile.dylib
    0x7fffdd748000 -     0x7fffdd760ff7  libsystem_asl.dylib (349.30.2) <EFAC72D7-CB13-3DF7-ADF3-EC6635C6F1EA> /usr/lib/system/libsystem_asl.dylib
    0x7fffdd761000 -     0x7fffdd761ff7  libsystem_blocks.dylib (67) <B8C3701D-5A91-3D35-999D-2DC8D5393525> /usr/lib/system/libsystem_blocks.dylib
    0x7fffdd762000 -     0x7fffdd7effef  libsystem_c.dylib (1158.30.7) <2F881962-03CB-3B9D-A782-D98C1BBA4E3D> /usr/lib/system/libsystem_c.dylib
    0x7fffdd7f0000 -     0x7fffdd7f3ffb  libsystem_configuration.dylib (888.30.2) <4FE3983C-E4ED-3939-A578-03AD29C99788> /usr/lib/system/libsystem_configuration.dylib
    0x7fffdd7f4000 -     0x7fffdd7f7fff  libsystem_coreservices.dylib (41.4) <1A572B9E-0C47-320F-8C64-7990D0A5FB5A> /usr/lib/system/libsystem_coreservices.dylib
    0x7fffdd7f8000 -     0x7fffdd810ff3  libsystem_coretls.dylib (121.31.1) <4676F06D-274D-31BE-B61C-4D7A4AEF4858> /usr/lib/system/libsystem_coretls.dylib
    0x7fffdd811000 -     0x7fffdd817fff  libsystem_dnssd.dylib (765.30.11) <DC708D84-ED7D-3936-B996-A67C66B8DDAA> /usr/lib/system/libsystem_dnssd.dylib
    0x7fffdd818000 -     0x7fffdd841ff7  libsystem_info.dylib (503.30.1) <9ED9121C-F111-3FAD-BC2F-C95DEE1C9362> /usr/lib/system/libsystem_info.dylib
    0x7fffdd842000 -     0x7fffdd864ff7  libsystem_kernel.dylib (3789.41.3) <B75B128C-7D7A-3318-91CD-82B5A69C5329> /usr/lib/system/libsystem_kernel.dylib
    0x7fffdd865000 -     0x7fffdd8acfe7  libsystem_m.dylib (3121.4) <266DB92B-A86F-3691-80FB-1B26AD73CFF3> /usr/lib/system/libsystem_m.dylib
    0x7fffdd8ad000 -     0x7fffdd8cbff7  libsystem_malloc.dylib (116.30.3) <F40DEE3B-386A-3529-A3F7-98117ED55BF4> /usr/lib/system/libsystem_malloc.dylib
    0x7fffdd8cc000 -     0x7fffdd923ffb  libsystem_network.dylib (856.30.16) <4AE368E9-605D-379D-B04C-2AC7455B8250> /usr/lib/system/libsystem_network.dylib
    0x7fffdd924000 -     0x7fffdd92dff3  libsystem_networkextension.dylib (563.30.15) <EB020B0C-7DF0-3EEF-8E3C-15DA3C01D687> /usr/lib/system/libsystem_networkextension.dylib
    0x7fffdd92e000 -     0x7fffdd937ff3  libsystem_notify.dylib (165.20.1) <E7FD3A7C-DD07-36E2-9FA4-7561F9F114DA> /usr/lib/system/libsystem_notify.dylib
    0x7fffdd938000 -     0x7fffdd940fe7  libsystem_platform.dylib (126.1.2) <3CA06D4E-C00A-36DE-AA65-3A390097D1F6> /usr/lib/system/libsystem_platform.dylib
    0x7fffdd941000 -     0x7fffdd94bff7  libsystem_pthread.dylib (218.30.1) <C869ED7C-BE29-3532-8E69-3A8DA1447EDC> /usr/lib/system/libsystem_pthread.dylib
    0x7fffdd94c000 -     0x7fffdd94fff7  libsystem_sandbox.dylib (592.31.1) <7BBFDF96-293F-3DD9-B3A4-7C168280B441> /usr/lib/system/libsystem_sandbox.dylib
    0x7fffdd950000 -     0x7fffdd951fff  libsystem_secinit.dylib (24) <5C1F1E47-0F7D-3E25-8DEB-D9DB1F902281> /usr/lib/system/libsystem_secinit.dylib
    0x7fffdd952000 -     0x7fffdd959fff  libsystem_symptoms.dylib (532.30.6) <5D990CF5-B58F-39F7-B375-99B4EC62CFBD> /usr/lib/system/libsystem_symptoms.dylib
    0x7fffdd95a000 -     0x7fffdd97aff7  libsystem_trace.dylib (518.30.7) <6D34D1EA-2A3C-3D2D-803E-A666E6AEEE52> /usr/lib/system/libsystem_trace.dylib
    0x7fffdd97b000 -     0x7fffdd980ffb  libunwind.dylib (35.3) <9F7C2AD8-A9A7-3DE4-828D-B0F0F166AAA0> /usr/lib/system/libunwind.dylib
    0x7fffdd981000 -     0x7fffdd9aaff7  libxpc.dylib (972.30.7) <65E41BB6-EBD5-3D93-B0BE-B190CEE4DD93> /usr/lib/system/libxpc.dylib

External Modification Summary:
  Calls made by other processes targeting this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by all processes on this machine:
    task_for_pid: 592380
    thread_create: 0
    thread_set_state: 0

VM Region Summary:
ReadOnly portion of Libraries: Total=120.0M resident=0K(0%) swapped_out_or_unallocated=120.0M(100%)
Writable regions: Total=17.4M written=0K(0%) resident=0K(0%) swapped_out=0K(0%) unallocated=17.4M(100%)
 
                                VIRTUAL   REGION 
REGION TYPE                        SIZE    COUNT (non-coalesced) 
===========                     =======  ======= 
Kernel Alloc Once                    8K        2 
MALLOC                            9396K        8 
MALLOC guard page                   16K        4 
STACK GUARD                       56.0M        2 
Stack                             8192K        2 
VM_ALLOCATE                          4K        2 
__DATA                            1572K       44 
__LINKEDIT                       111.6M        5 
__TEXT                            8616K       43 
shared memory                        8K        3 
===========                     =======  ======= 
TOTAL                            194.7M      105 

```
</details>

**Comment by elyscape** [comment]
Another weird data point: if I use the binary from the releases page, it works fine:
```shellsession
scapeless:~ elyscape$ echo "$VAL" | ./jq-osx-amd64 'match(".*")'
{
  "offset": 0,
  "length": 200,
  "string": "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
  "captures": []
}
```
I've been using [the version from Homebrew](https://github.com/Homebrew/homebrew-core/blob/master/Formula/jq.rb).

**Comment by pkoppstein** [comment]
Yes, that's it. I was using /usr/local/Cellar/oniguruma/5.9.5

After "upgrading" to 6.1.3 (brew upgrade oniguruma) , bingo!

Maybe something changed in the interface?

**Comment by wtlangford** [comment]
I doubt it's the interface, given that backtrace. I can take a look, but I
don't know how much we can do. It might be an oniguruma issue.

**Comment by nicowilliams** [comment]
It sure smells like a problem with Oniguruma.

**Comment by elyscape** [comment]
The really weird thing is that Fedora 25 has Oniguruma 6.1.3 (current) but jq doesn't blow up there:
```shellsession
[root@0245a78dda77 /]# cat /etc/fedora-release 
Fedora release 25 (Twenty Five)
[root@0245a78dda77 /]# dnf install -y jq
Last metadata expiration check: 0:02:59 ago on Wed Mar 29 18:46:38 2017.
Dependencies resolved.
================================================================================
 Package           Arch           Version                 Repository       Size
================================================================================
Installing:
 jq                x86_64         1.5-5.fc25              updates         154 k
 oniguruma         x86_64         6.1.3-1.fc25            updates         164 k

Transaction Summary
================================================================================
Install  2 Packages

Total download size: 318 k
Installed size: 1.1 M
Downloading Packages:
(1/2): jq-1.5-5.fc25.x86_64.rpm                 205 kB/s | 154 kB     00:00    
(2/2): oniguruma-6.1.3-1.fc25.x86_64.rpm        216 kB/s | 164 kB     00:00    
--------------------------------------------------------------------------------
Total                                            50 kB/s | 318 kB     00:06     
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Installing  : oniguruma-6.1.3-1.fc25.x86_64                               1/2 
  Installing  : jq-1.5-5.fc25.x86_64                                        2/2 
  Verifying   : jq-1.5-5.fc25.x86_64                                        1/2 
  Verifying   : oniguruma-6.1.3-1.fc25.x86_64                               2/2 

Installed:
  jq.x86_64 1.5-5.fc25               oniguruma.x86_64 6.1.3-1.fc25              

Complete!
[root@0245a78dda77 /]# (echo -n '"' ; for num in {1..10000} ; do echo -n 'a' ; done ; echo '"') | jq 'match(".*")'
{
  "offset": 0,
  "length": 10000,
  "string": "[elided]",
  "captures": []
}
```
And it doesn't appear to be statically compiled:
```shellsession
[root@0245a78dda77 /]# ldd /usr/bin/jq
	linux-vdso.so.1 (0x00007ffea08aa000)
	libjq.so.1 => /usr/lib64/libjq.so.1 (0x00007f08142e1000)
	libm.so.6 => /usr/lib64/libm.so.6 (0x00007f0813fd8000)
	libonig.so.4 => /usr/lib64/libonig.so.4 (0x00007f0813d37000)
	libc.so.6 => /usr/lib64/libc.so.6 (0x00007f0813971000)
	/lib64/ld-linux-x86-64.so.2 (0x00005585bb055000)
```

**Comment by elyscape** [comment]
Actually, looks like it does happen on Fedora 25, but only occasionally:
```shellsession
[root@09c13459a97f ~]# VAL="$(echo -n '"' ; for num in {1..200} ; do echo -n 'a' ; done ; echo '"')"
[root@09c13459a97f ~]# while true ; do echo "$VAL" | jq 'match(".*")' ; echo ; sleep 1 ; done
{
  "offset": 0,
  "length": 200,
  "string": "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
  "captures": []
}

{
  "offset": 0,
  "length": 200,
  "string": "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
  "captures": []
}

Segmentation fault

{
  "offset": 0,
  "length": 200,
  "string": "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
  "captures": []
}

Segmentation fault

{
  "offset": 0,
  "length": 200,
  "string": "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
  "captures": []
}

Segmentation fault

{
  "offset": 0,
  "length": 200,
  "string": "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
  "captures": []
}

{
  "offset": 0,
  "length": 200,
  "string": "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
  "captures": []
}
```
So that's weird.

**Comment by wtlangford** [comment]
Yeah, this is an oniguruma bug for sure. It reproduces in a minimal case
quite easily. I'll file it with them later tonight.

**Comment by wtlangford** [comment]
Issue filed with oniguruma: https://github.com/kkos/oniguruma/issues/52

**Comment by wtlangford** [comment]
This was fixed on their master.  I've updated our submodule to fix it, but there's nothing I can do for the version in your OS' package manager.  Standalone binary releases of jq will link statically against oniguruma, of course.  Additionally, you can already build a working jq yourself: 
```
# If you don't have a jq cloned:
git clone https://github.com/stedolan/jq --recursive
# If you already have jq cloned, do this instead:
git submodule update --init
```

Then do this to build using the vendored copy:
```
autoreconf -fi
./configure --with-oniguruma=builtin
make
```
